app/controllers/application_controller.rb

```diff
@@ -25,15 +26,16 @@ def check_maintenance_mode_is_enabled
     end
   end
 
-  def append_info_to_payload(payload)
-    super
-    payload[:host] = request.host
-    payload[:request_id] = request.request_id
-    payload[:form_id] = params[:form_id] if params[:form_id].present?
-    payload[:page_id] = params[:page_slug] if params[:page_slug].present? && params[:page_slug].match(Page::PAGE_ID_REGEX)
-    payload[:page_slug] = params[:page_slug] if params[:page_slug].present?
-    payload[:session_id_hash] = session_id_hash
-    payload[:trace_id] = request.env["HTTP_X_AMZN_TRACE_ID"].presence
+  def set_logging_context
+    @logging_context = {}.tap do |h|
+      h[:host] = request.host
+      h[:request_id] = request.request_id
+      h[:form_id] = params[:form_id] if params[:form_id].present?
+      h[:page_id] = params[:page_slug] if params[:page_slug].present? && params[:page_slug].match(Page::PAGE_ID_REGEX)
+      h[:page_slug] = params[:page_slug] if params[:page_slug].present?
+      h[:session_id_hash] = session_id_hash
+      h[:trace_id] = request.env["HTTP_X_AMZN_TRACE_ID"].presence
```

Comment:
I kind of mangled this line when I added it, it probably should be

```suggestion
      h[:trace_id] = request.env["HTTP_X_AMZN_TRACE_ID"] if request.env["HTTP_X_AMZN_TRACE_ID"].present?
```

Reply:
yeah I think as it was it would have just set the `trace_id:` to `nil` but I've changed it as you suggest to omit the field if its not available instead.